nemo/export/quantize/quantizer.py

```diff
@@ -167,80 +98,73 @@ def dummy():
         set_data_parallel_group(mpu.get_data_parallel_group())
         set_tensor_parallel_group(mpu.get_tensor_model_parallel_group())
 
+        model.set_inference_config(OmegaConf.to_container(inference_config))
+
     @staticmethod
-    def _load_and_modify_config(
-        model_file: str,
-        tensor_model_parallel_size: Optional[int] = None,
-        pipeline_model_parallel_size: Optional[int] = None,
-    ):
-        model_cfg = load_config(model_file)
-
-        with open_dict(model_cfg):
-            model_cfg.activations_checkpoint_method = None
-            model_cfg.activations_checkpoint_granularity = None
-            model_cfg.sequence_parallel = False
-            if tensor_model_parallel_size is not None:
-                model_cfg.tensor_model_parallel_size = tensor_model_parallel_size
-            if pipeline_model_parallel_size is not None:
-                model_cfg.pipeline_model_parallel_size = pipeline_model_parallel_size
+    def modify_model_config(model_config: DictConfig) -> DictConfig:
+        with open_dict(model_config):
+            model_config.activations_checkpoint_method = None
+            model_config.activations_checkpoint_granularity = None
+            model_config.sequence_parallel = False
             # Only custom ModelOpt spec is supported for PTQ: this custom spec is largely based on local Megatron-LM
             # layer definitions to avoid Transformer Engine implementations that are currently not supported.
             # This layer spec also requires RoPE fusion to be disabled for tensor view operations in attention
             # layer implementation from megatron/core/transformer/dot_product_attention.py to be functional.
-            model_cfg.name = "modelopt"
-            model_cfg.apply_rope_fusion = False
+            model_config.name = "modelopt"
+            model_config.apply_rope_fusion = False
 
-        return model_cfg
+        return model_config
 
     @staticmethod
-    def _sample_output(model):
-        """Generate sample output for a model instance."""
-        if torch.distributed.get_rank() == 0:
-            print("Generating sample output for a model...")
-
-        response = model.generate(
-            inputs=[
-                "Born in north-east France, Soyer trained as a",
-                "Born in California, Soyer trained as a",
-            ],
-            length_params={
-                "max_length": 100,
-                "min_length": 100,
-            },
-        )
-
-        if torch.distributed.get_rank() == 0:
-            print(f'Example NeMo output after PTQ: {response["sentences"]}"')
-
     def quantize(
```

Comment:
Different quantization pipelines may have different way of using dataloader or writing the forward loop hence changed the arguments to take  forward loop directly instead of dataloader (mtq.quantize also requires forward loop from a user instead of dataloader).

Reply:
This would be my major remark in general, other things should be more or less clear

Reply:
The only issue with that is `inference_config` and `export_config` are optional for QAT and we only need `quantization_config` in Quantizer. How do you feel about passing `quantization_config` to `Quantizer.__init__` and then `export_config` will be an argument to `Quantizer.export()` and we can remove `inference_config` all together as it is not required as I suggested in another comment?

Reply:
I like the proposal, this is what I also was thinking of here https://github.com/kevalmorabia97/NeMo/compare/kmorabia/update-quantizer-for-qat...janekl:NeMo:review

Removing `inference_config` and using `forward_bwd_step` sounds good but we'll need to run accuracy check. There are two options:
* run this with [nemo_example.sh](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/llm_ptq/scripts/nemo_example.sh)in the source modelopt repo
* run Lambada with [test_nemo_export.py](https://github.com/NVIDIA/NeMo/blob/main/tests/export/test_nemo_export.py) on quantized checkpoints (with a new quantized checkpoint from megatron_quantization.py after changes) - that one is also a part of internal CI